test/Microsoft.AspNetCore.Mvc.IntegrationTests/SpecialBindingSourceMetadataProviderIntegrationTest.cs

```diff
@@ -0,0 +1,152 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Internal;
+using Microsoft.AspNetCore.Mvc.Abstractions;
+using Microsoft.AspNetCore.Mvc.Internal;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+using Microsoft.AspNetCore.Mvc.ModelBinding.Binders;
+using Microsoft.Extensions.Primitives;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Mvc.IntegrationTests
+{
+    public class SpecialBindingSourceMetadataProviderIntegrationTest
+    {
+        [Fact]
+        public async Task BindParameter_WithCancellationToken_BindingSourceSpecial()
+        {
+            // Arrange
+            var options = new MvcOptions();
+            var setup = new MvcCoreMvcOptionsSetup(new TestHttpRequestStreamReaderFactory());
+
+            options.ModelBinderProviders.Insert(0, new CancellationTokenModelBinderProvider());
+
+            setup.Configure(options);
+
+            var argumentBinder = ModelBindingTestHelper.GetArgumentBinder(options);
+            var parameter = new ParameterDescriptor()
+            {
+                Name = "Parameter1",
+                BindingInfo = new BindingInfo(),
+                ParameterType = typeof(CancellationTokenBundle),
+            };
+
+            var testContext = ModelBindingTestHelper.GetTestContext(request =>
+            {
+                request.Form = new FormCollection(new Dictionary<string, StringValues>
+                {
+                    { "name", new[] { "Fred" } }
+                });
+            });
+
+            var modelState = testContext.ModelState;
+            var token = testContext.HttpContext.RequestAborted;
+
+            // Act
+            var modelBindingResult = await argumentBinder.BindModelAsync(parameter, testContext);
+
+            // Assert
+            // ModelBindingResult
+            Assert.True(modelBindingResult.IsModelSet);
+
+            // Model
+            var boundPerson = Assert.IsType<CancellationTokenBundle>(modelBindingResult.Model);
+            Assert.NotNull(boundPerson);
+            Assert.Equal("Fred", boundPerson.Name);
+            Assert.Equal(token, boundPerson.Token);
+
+            // ModelState
+            Assert.True(modelState.IsValid);
+        }
```

Comment:
Is this just a test that the binding source has no effect on the runtime behavior?